Sources/GRPCReflectionService/Documentation.docc/ReflectionServiceTutorial.md

```diff
@@ -0,0 +1,235 @@
+```
+
+Let's break the command down:
+ - The first argument that we are passing to `protoc` is the path 
+to the proto file we want to generate the serialized file descriptor proto
+for: [Sources/Examples/HelloWorld/Model/helloworld.proto][helloworld-proto].
+- The `proto_path` flag is the path to search for imports: 
+Sources/Examples/HelloWorld/Model.
+- The plugin we are using is `./.build/debug/protoc-gen-grpc-swift`.
+- The options for the `grpc-swift_opt` flag that we should set are:
+`Client=false,Server=false`, because we don't want to generate code
+and `ReflectionData=true` to signal that we want to generate a
+file containing the serialized file descriptor proto.
+- The `grpc-swift_out` flag is used to set the path of the directory
+where the generated file will be located in: Sources/Examples/ReflectionService/Generated.
+
+The commands for generating the binary files representing the serialized file descriptor 
+protos for the `Echo` and `Reflection` services are similar. The path of the output
+directory, the plugin and the options are the same. The path to search for imports
+is the parent directory of the proto file.
+
+The paths of the proto files are: 
+- [Sources/Examples/Echo/Model/echo.proto][echo-proto]
+- [Sources/GRPCReflectionService/v1/reflection-v1.proto][v1]
+
+Depending on the version of gRPCurl you are using you might need to use the v1alpha proto instead.
+
+ ### Instantiating the Reflection Service 
+
+ In the Server implementation, we have to instantiate each provider.
+ For instantiating the `ReflectionService` provider we have to pass in an array
+ of Strings representing the paths to the binary files we have just generated,
+withing the project and the version of the reflection, in our case v1.
+
+
+```swift
+let paths = [
+  "Sources/Examples/ReflectionService/Generated/helloworld.grpc.reflection.txt", 
+  "Sources/Examples/ReflectionService/Generated/echo.grpc.reflection.txt", 
+  "Sources/Examples/ReflectionService/Generated/reflection.grpc.reflection.txt"
+  ]
+
+let reflectionServiceProvider = try ReflectionService(
+  serializedFileDescriptorProtoFilePaths: paths, version: .v1
+)
+```
+
+### Running the Server
+
+In our example the server is not configured with TLS. The port is `1234`.
+Starting the server:
+
+```swift
+let server = try await Server.insecure(group: group)
+  .withServiceProviders([reflectionServiceProvider, GreeterProvider(), EchoProvider()])
+  .bind(host: "localhost", port: self.port)
+  .get()
+
+```
+
+To run the server, from the root of the package run:
+
+```sh
+$ swift run ReflectionServiceServer
+```
+
+ ## Testing the Reflection Service using GRPCurl
+
+Please follow the instructions from the [GRPCurl README][grpcurl-setup] in order to set gRPCurl up.
+
+From a different terminal than the one used for running the Server, we will call gRPCurl commands,
+following the format:
+
+`grpcurl [flags] [address] [list|describe] [symbol]`.
```

Comment:
Use single backticks if the code is inline, i.e. within a sentence. Otherwise use triple backticks

```suggestion
following the format: `grpcurl [flags] [address] [list|describe] [symbol]`.
```